### DISCUSSION TITLE
Is there a way to get the time?

### DISCUSSION THREAD

**Discussion by mightyiam** [remotemobprogramming/mob#273] [General]
There are two sources of truth for time. There's a local `sleep N && ...` I noticed in the process list. And there's the web time. Is there a way to get any of those using the CLI?

**Comment by simonharrer** [comment]
Currently, there is no way. What would you like to have?

**Reply by mightyiam** [reply]
I'd like to have the time on my status bar. By status bar, I mean in my case https://github.com/greshake/i3status-rust.
It supports custom widgets that are implemented using arbitrary external commands.

**Reply by simonharrer** [reply]
You can connect to a room at timer.mob.sh similar to the macos appstore widget or the website itself. It uses serverside events. If you want I can describe the interface so you can code against it. And if this doesn’t suffice, we could think of adding an additional event or an additional api that helps you. How would the perfect api look to you when integrating with i3status?

You cannot show the remaining local timer time, that’s currently not possible.

**Reply by gregorriegler** [reply]
It would be solved once we have the hooks. Then you can just script it easy peasy. On mob start X -> Send start timer command to your timer.
And I would do this as well with my https://github.com/gregorriegler/polybar-egg-timer

**Reply by simonharrer** [reply]
User stories:

- start a new timer when you start a new timer
- start a new timer when someone else start a new timer
- override an existing timer when you started a new timer
- override an existing timer when someone else started a new timer

With local hooks, you'll only be able to handle the timers you start yourself, but not the ones that others start.

With a listener to the server sent events from a room at timer.mob.sh, you could handle all user stories accordingly. But it requires some logic in between that listens to these events and forwards the infos to the local timer displays accordingly.

**Reply by mightyiam** [reply]
I'd be happy with having the current time on my shell prompt. I'd be okay that this is available only while I'm the driver. I'm wondering whether it would make sense that the mob tool would have a user systemd service for any process to query...
